File: less Comfortable/pset1/greedy/greedy.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main (void)
{
    float m;
    
    do
    {
        printf("O hai! How much change is owed?\n");
        m = get_float();
    }
    while (m < 0);
    
    int i = round(100 * m);
    
    int a = remainder (i , 25);
    if (a < 0)
        a = (a + 25);
    int w = ((i - a) / 25);

    int b = remainder (a , 10);
    if (b < 0)
        b = (b + 10);
    int x = ((a - b) /10);
    
    int c = remainder (b , 5);
    if (c < 0)
        c = (c + 5);
    int y = ((b - c) /5);

    printf("%i\n", w + x + y + c); 
}

File: less Comfortable/pset6/greedy.py
import cs50

# get the change owed
while True:
    print("O hai! How much change is owed?")
    m = cs50.get_float()
    if m > 0:
        break
    
# convert from dollars to cents 
i = round(100 * m)

# get remainder from 25
# that is the amount that has to be payed using
# coins of lower denominations
a = i % 25
# get the number of 25 cent coins used
w = ((i - a) / 25)

# repeat the above process for other denominations

b = a % 10
# get the number of 10 cent coins used
x = ((a - b) / 10)

c = b % 5
# get the number of 5 cent coins used
# also, c is the number of 1 cents to be paid
y = ((b - c) / 5)
    
print("{}".format(w + y + x + c))
    
    
    
    
    
    